src/content_cao.cpp

```diff
@@ -667,6 +672,8 @@ void GenericCAO::processInitData(const std::string &data)
 GenericCAO::~GenericCAO()
 {
 	if (m_is_player) {
+		if (m_client->moddingEnabled() && !m_client->isShutdown())
+			m_client->getScript()->on_leaveplayer(m_name);
```

Comment:
this is a little bit incorrect as everything is a CAO client side, you distringuish players from other entities